src/lib/telemetry/TelemetryReporter.ts

```diff
@@ -140,23 +141,40 @@ export class TelemetryReporter {
       return error
     })
     await ensureDir(join(TelemetryReporter.PENDING_DATA_DIR, 'errors'))
-    await ensureFile(metaErrorFilePath)
     await writeJson(metaErrorFilePath, errorsReport)
   }
 
-  public async ensureTelemetryDirMazSize() {
-    const telemetryFiles = await readdirRecursive(TelemetryCollector.TELEMETRY_LOCAL_DIR)
+  public async ensureTelemetryDirMaxSize() {
+    const telemetryFiles = await glob('**', { cwd: TelemetryCollector.TELEMETRY_LOCAL_DIR, absolute: true })
     let telemetryDirSize = 0
-    await Promise.all(telemetryFiles.map(async (file) => {
-      const fileStats = await stat(file)
-      telemetryDirSize += fileStats.size
-    }))
-    if (telemetryDirSize > 10 * 1000 * 1000) {
+    await Promise.all(
+      telemetryFiles.map(async file => {
+        const fileStats = await stat(file)
+        telemetryDirSize += fileStats.size
+      })
+    )
+    if (telemetryDirSize > TelemetryReporter.MAX_TELEMETRY_DIR_SIZE) {
       await remove(TelemetryCollector.TELEMETRY_LOCAL_DIR)
     }
   }
 
-  private async treatReportError(reportError, reportType: 'metrics' | 'errors', metricsOrErrors: MetricReport[] | ErrorReport[]) {
+  private async treatReportError(
+    reportError,
+    reportType: 'metrics',
+    metricsOrErrors: MetricReport[]
```

Comment:
This can be `metrics`
```suggestion
    metrics: MetricReport[]
```